### PR TITLE
NCD-672: initial code review

### PR DESCRIPTION
**Pull request by cybic**
Tried to incorporate as much as possible from the code review.

### CODE REVIEW

File: src/SidePanel.tsx [hunk 1]
```diff
@@ -30,9 +30,9 @@ export default () => {
                 <Button
                     disabled={!device}
                     variant="primary"
-                    className="w-100"
+                    className="tw-w-full"
                     onClick={() => {
-                        console.dir(device);
+                        // console.dir(device);
```
Comment (aadnekar) [inline src/SidePanel.tsx:35]: Why do you not remove the console.log?

File: src/features/Configuration/Configuration.tsx [hunk 2]
```diff
@@ -52,101 +54,98 @@ const BoardController: React.FC<{ active: boolean }> = ({ active }) => {
...
+        logger.debug(
+            `Got device ${JSON.stringify(device)} ${JSON.stringify(
+                boardRevision
+            )}`
+        );
+        logger.debug(
+            `Device with boardVersion: ${JSON.stringify(
+                device?.devkit?.boardVersion
+            )}`
         );
```
Comment (aadnekar) [inline src/features/Configuration/Configuration.tsx:66]: If the plan is to keep this, we could consider extracting the logging into a function?

```typescript
// Define it somewhere:
const logDeviceInfo = (device: Device) => {
     logger.debug(
            `Got device ${JSON.stringify(device)} ${JSON.stringify(
                boardRevision
            )}`
        );
        logger.debug(
            `Device with boardVersion: ${JSON.stringify(
                device?.devkit?.boardVersion
            )}
}

// ... and call it
logDeviceInfo(device);
```
sorry for the formatting, wish Github had autoformatting

**Review by aadnekar — APPROVED** [review]
Great improvements!

I really think it's cool to see the `xor` function in use, think it helps to make the code consice and readable.
We could considering extracting it into it's own file, and then rather importing it in the different use cases. But that's far from important.

Well done! 🙌

File: src/SidePanel.tsx (hunk 1 above)
Comment (cybic) [reply src/SidePanel.tsx:35]: I had actually removed it but for some reason it wasn't committed :/ Doing it now :+1:

**Pull request merged**